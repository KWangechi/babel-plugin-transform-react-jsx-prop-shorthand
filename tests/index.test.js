import { describe, it, expect } from "vitest";
import fs from "fs";
import path from "path";
import { build } from "vite";
import react from "@vitejs/plugin-react";
import fg from "fast-glob";
import {jsxTransformPropShorthand} from "../src/index.js";
import { transform } from "@babel/core";

describe("Fixtures Tests", async () => {
  const fixturesDir = path.resolve(__dirname, "fixtures");

  const subfolders = fs.readdirSync(fixturesDir).filter((file) => {
    return fs.statSync(path.join(fixturesDir, file)).isDirectory();
  });
  it("should ensure all fixture subfolders exist", () => {
    expect(fs.existsSync(fixturesDir)).toBe(true);

    expect(subfolders.length).toBeGreaterThan(0);
  });

  it("should ensure each fixture subfolder contains files", () => {
    subfolders.forEach((subfolder) => {
      const files = fs.readdirSync(path.join(fixturesDir, subfolder));
      expect(files.length).toBeGreaterThan(0);
    });
  });

  // it("should transform and bundle .jsx files correctly", async () => {
  //   const fixturesPath = path.resolve(__dirname, "fixtures");
  //   const jsxFiles = await fg("**/*.{jsx,tsx}", { cwd: fixturesPath });


  //   for (const file of jsxFiles) {
  //     const filePath = path.join(fixturesPath, file);
  //     const fileName = path.basename(file, path.extname(file));

  //     const result = await build({
  //       root: path.resolve(__dirname, ".."),
  //       build: {
  //         outDir: "dist", 
  //         rollupOptions: {
  //           input: filePath,
  //           external: ["react", "react-dom"],
  //           output: {
  //             entryFileNames: `${fileName}.jsx`,
  //             dir: "dist",
  //             format: "es",
  //           },
  //         },
  //       },
  //       plugins: [
  //         react({
  //           babel: {
  //             plugins: [jsxTransformPropShorthand()],
  //           },
  //         }),
  //       ],
  //     });

  //     console.log(`Built file: ${fileName}.js`);
  //     expect(result).toBeDefined();
  //   }
  // });

  it("should transform shorthand props correctly", () => {
    const inputCode = `
    const propA = 'Hello World';
    <ChildComp ::propA />`;

    const result = transform(inputCode, {
      presets: ["@babel/preset-react"],
      plugins: [jsxTransformPropShorthand()],
    });

    expect(result.code).toMatchSnapshot();
  });

  // test to see if it bundles correctly with Vite
  // it("should transform and bundle JSX correctly", async () => {
  //   const fixturesPath = path.resolve(__dirname, "fixtures/react");
  //   const inputFile = path.join(fixturesPath, "App.jsx");
  //   const outputFile = path.resolve(__dirname, "../dist/App.js");

  //   // Run the Vite build
  //   await build({
  //     root: path.resolve(__dirname, ".."),
  //     build: {
  //       outDir: "dist",
  //       rollupOptions: {
  //         input: inputFile,
  //       },
  //     },
  //     plugins: [
  //       react({
  //         babel: {
  //           plugins: [jsxTransformPropShorthand()],
  //         },
  //       }),
  //     ],
  //   });

  //   // Read the output file
  //   const outputCode = fs.readFileSync(outputFile, "utf-8");

  //   // Assert the transformed output
  //   expect(outputCode).toContain(`<ChildComp propA={propA} />`);
  // });
});
