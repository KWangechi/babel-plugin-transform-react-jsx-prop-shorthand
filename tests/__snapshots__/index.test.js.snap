// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fixtures Tests > should bundle  1`] = `
"const propA = 'Hello World';
/*#__PURE__*/React.createElement(ChildComp, {
  propA: propA
});"
`;

exports[`Fixtures Tests > should bundle correctly with Vite 1`] = `
"const propA = 'Hello World';
/*#__PURE__*/React.createElement(ChildComp, {
  propA: propA
});"
`;

exports[`Fixtures Tests > should process .tsx and .jsx files correctly 1`] = `
{
  "output": [
    {
      "code": "
",
      "dynamicImports": [],
      "exports": [],
      "facadeModuleId": "C:/Users/user/Documents/WANGECHI'S DOCUMENTS/PERSONAL PROJECTS/DUMB PROJECTS/babel-plugin-transform-react-jsx-prop-shorthand/tests/fixtures/func-scope/App.jsx",
      "fileName": "assets/App-l0sNRNKZ.js",
      "implicitlyLoadedBefore": [],
      "importedBindings": {},
      "imports": [],
      "isDynamicEntry": false,
      "isEntry": true,
      "isImplicitEntry": false,
      "map": null,
      "moduleIds": [
        "C:/Users/user/Documents/WANGECHI'S DOCUMENTS/PERSONAL PROJECTS/DUMB PROJECTS/babel-plugin-transform-react-jsx-prop-shorthand/tests/fixtures/func-scope/App.jsx",
      ],
      "modules": {
        "C:/Users/user/Documents/WANGECHI'S DOCUMENTS/PERSONAL PROJECTS/DUMB PROJECTS/babel-plugin-transform-react-jsx-prop-shorthand/tests/fixtures/func-scope/App.jsx": {
          "code": null,
          "originalLength": 0,
          "removedExports": [],
          "renderedExports": [],
          "renderedLength": 0,
        },
      },
      "name": "App",
      "preliminaryFileName": "assets/App-!~{001}~.js",
      "referencedFiles": [],
      "sourcemapFileName": null,
      "type": "chunk",
      "viteMetadata": {
        "importedAssets": Set {},
        "importedCss": Set {},
      },
    },
  ],
}
`;

exports[`Fixtures Tests > should process .tsx and .jsx files correctly 2`] = `
{
  "output": [
    {
      "code": "
",
      "dynamicImports": [],
      "exports": [],
      "facadeModuleId": "C:/Users/user/Documents/WANGECHI'S DOCUMENTS/PERSONAL PROJECTS/DUMB PROJECTS/babel-plugin-transform-react-jsx-prop-shorthand/tests/fixtures/global-imports/App.jsx",
      "fileName": "assets/App-l0sNRNKZ.js",
      "implicitlyLoadedBefore": [],
      "importedBindings": {},
      "imports": [],
      "isDynamicEntry": false,
      "isEntry": true,
      "isImplicitEntry": false,
      "map": null,
      "moduleIds": [
        "C:/Users/user/Documents/WANGECHI'S DOCUMENTS/PERSONAL PROJECTS/DUMB PROJECTS/babel-plugin-transform-react-jsx-prop-shorthand/tests/fixtures/global-imports/App.jsx",
      ],
      "modules": {
        "C:/Users/user/Documents/WANGECHI'S DOCUMENTS/PERSONAL PROJECTS/DUMB PROJECTS/babel-plugin-transform-react-jsx-prop-shorthand/tests/fixtures/global-imports/App.jsx": {
          "code": null,
          "originalLength": 0,
          "removedExports": [],
          "renderedExports": [],
          "renderedLength": 0,
        },
      },
      "name": "App",
      "preliminaryFileName": "assets/App-!~{001}~.js",
      "referencedFiles": [],
      "sourcemapFileName": null,
      "type": "chunk",
      "viteMetadata": {
        "importedAssets": Set {},
        "importedCss": Set {},
      },
    },
  ],
}
`;

exports[`Fixtures Tests > should transform shorthand props correctly 1`] = `
"const propA = 'Hello World';
/*#__PURE__*/React.createElement(ChildComp, {
  propA: propA
});"
`;
